spec/unit/widgets/checkbox_spec.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require 'rspec'
+
+module LibyuiClient
+  module Widgets
+    RSpec.describe Checkbox do
+      let(:widget_controller) { instance_double('widget_controller') }
+      let(:widget_filter) { { id: 'change_now' } }
+      subject { Checkbox.new(widget_controller, widget_filter) }
+      describe '#toggle' do
+        it 'has a request with action toggle' do
+          # logger = object_double("LibyuiClient::logger", :info => nil).as_stubbed_const
+          allow(widget_controller).to receive(:send_action)
+          expect(subject).to receive(:action).with({ action: 'toggle' })
+          expect(subject.toggle).to be_instance_of(Checkbox)
```

Comment:
As per our discussion, let's put check for the returned result to validate that `self` is returned and not that it's instance of given class.

Reply:
i could not find a way to check for the specific instance but i can check that it is not `nil`. if you remove the `self` from the methods the test should fail now.